Bottom: af4afd3216b9addcb5809a68e568b3213b6c03a1
Top:    25b370973988353131f3b8ea53234d10e60eda64
Author: Ian Kent <raven@themaw.net>
Date:   2013-08-19 09:57:34 +0800

debug.patch

At the moment the internal hosts map can only be an entry in the master
map but there's no reason it shouldn't be possible to use it in map
entries.


---

diff --git a/lib/parse_subs.c b/lib/parse_subs.c
index 2326838..cc1b4a4 100644
--- a/lib/parse_subs.c
+++ b/lib/parse_subs.c
@@ -35,6 +35,7 @@ static struct types map_type[] = {
 	{ "ldaps", 5 },
 	{ "hesiod", 6 },
 	{ "userdir", 7 },
+	{ "hosts", 5 },
 };
 static unsigned int map_type_count = sizeof(map_type)/sizeof(struct types);
 
diff --git a/modules/parse_sun.c b/modules/parse_sun.c
index c1fc528..9112918 100644
--- a/modules/parse_sun.c
+++ b/modules/parse_sun.c
@@ -881,7 +881,8 @@ static int validate_location(unsigned int logopt, char *loc)
 		    !strncmp(ptr, "file:", 5) || !strncmp(ptr, "yp:", 3) ||
 		    !strncmp(ptr, "nis:", 4) || !strncmp(ptr, "nisplus:", 8) ||
 		    !strncmp(ptr, "ldap:", 5) || !strncmp(ptr, "ldaps:", 6) ||
-		    !strncmp(ptr, "sss:", 4) || !strncmp(ptr, "dir:", 4))
+		    !strncmp(ptr, "sss:", 4) || !strncmp(ptr, "dir:", 4) ||
+		    !strncmp(ptr, "hosts:", 4))
 			return 1;
 		error(logopt,
 		      "expected colon delimeter not found in location %s",
@@ -1292,6 +1293,15 @@ int parse_mount(struct autofs_point *ap, const char *name,
 		return 1;
 	}
 
+	/*
+	 * Transform hosts map into a map type so we don't confuse
+	 * the options parsing.
+	 */
+	if (strstr(pmapent, "-hosts")) {
+		char *t = strstr(pmapent, "-hosts");
+		*t++ = 'h'; *t++ = 'o'; *t++ = 's'; *t++ = 't'; *t++ = 's';
+		*t = ':';
+	}
 	p = skipspace(pmapent);
 
 	/* Deal with 0 or more options */
